src/main/java/backend/RepoId.java

```diff
@@ -0,0 +1,100 @@
+package backend;
+
+import org.eclipse.egit.github.core.RepositoryId;
+import util.Utility;
+
+/**
+ * Represents the ID of a repository. To be used to ensure safe comparison
+ */
+public final class RepoId {
+
+    private final String repoOwner;
+    private final String repoName;
+
+    /**
+     * Constructor
+     * @param repoIdString repoOwner and repoName separated by slash
+     * @throws IllegalArgumentException If repoIdString is invalid
+     */
+    public RepoId(String repoIdString) throws IllegalArgumentException {
+        boolean isWellFormed = RepoId.isWellFormedRepoIdString(repoIdString);
+        throwExceptionIfNotWellFormed(isWellFormed);
+
+        String[] repoIDComponents = repoIdString.toLowerCase().split("/");
+        this.repoOwner = Utility.removeAllWhitespace(repoIDComponents[0]);
+        this.repoName = Utility.removeAllWhitespace(repoIDComponents[1]);
+    }
+
+    /**
+     * Constructor
+     * @param repoOwner Owner of the repo
+     * @param repoName Name of the repo
+     * @throws IllegalArgumentException If repoOwner or repoName is invalid
+     */
+    public RepoId(String repoOwner, String repoName) throws IllegalArgumentException {
+        boolean isWellFormed = RepoId.isWellFormedRepoIdString(repoOwner, repoName);
+        throwExceptionIfNotWellFormed(isWellFormed);
+
+        this.repoOwner = Utility.removeAllWhitespace(repoOwner.toLowerCase());
+        this.repoName = Utility.removeAllWhitespace(repoName.toLowerCase());
+
+    }
+
+    /**
+     * Check if repoOwner and repoName is non-null and non-empty
+     * and able to generate a well formed repoIdString of form <repoOwner>/<repoName>
+     * @param repoOwner Owner of the repo
+     * @param repoName Name of the repo
+     * @return True if the above conditions are met else False is returned
+     */
+    public static boolean isWellFormedRepoIdString(String repoOwner, String repoName) {
+        return !(repoOwner == null || repoOwner.isEmpty() || repoName == null || repoName.isEmpty())
+                && isWellFormedRepoIdString(RepositoryId.create(repoOwner, repoName).generateId());
+    }
+
+    /**
+     * Check if repoIdString non-null, non-empty
+     * and is of form <repoOwner>/<repoName>
+     * @param repoIdString
+     * @return True if the above conditions are met else False is returned
+     */
+    public static boolean isWellFormedRepoIdString(String repoIdString) {
+        RepositoryId repositoryId = RepositoryId.createFromId(repoIdString);
+        return repoIdString != null && !repoIdString.isEmpty() && repositoryId != null
+                && repoIdString.equals(repositoryId.generateId());
+    }
```

Comment:
Again, get rid of one of the overloaded methods to reduce code and reduce duplication?